### ISSUE TITLE
Improve documentation on control sample in reference and vignette

### ISSUE THREAD

**Issue by twoshoes** [ArnaudDroitLab/metagene2#21]
Hi,

I've been using metagene2 with control samples labelled as 2 in my experimental design. I was experimenting with different controls and noticed that changing the control makes no difference to the reported coverage, and neither does removing the control completely. 

I was wondering if I was doing something wrong, or what is the role of the samples labelled as control?

Thanks!

**Comment by ericfournier2** [comment]
Hello twoshoes,

here is how control samples interact with the metagene2 pipeline:
1. Whether or not control samples are present, coverages for all bams are still calculated individually. These are what you will get when calling $get_coverages() or $get_normalized_coverages() methods.
2. When the $group_coverages() method is run, whether because you called it directly or because it gets called as part of $produce_metagenes() or any other downstream method, the processing of control samples depends on your chosen normalization method:
- If you chose NULL (raw) or "RPM", control samples are ignored
- If you chose "NCIS", then the [NCIS algorithm](https://www.ncbi.nlm.nih.gov/pubmed/22883957) is used to subtract control coverages from treatment/input coverages
- If you chose log2_ratio (devel version only), then the ratio of the sum of all treatments vs the sum of all controls is calculated.
Those control-corected coverages are then returned from the $group_coverages() method.

which normalization scheme are you using? Unless you choose NCIS (or log-ratio on the devel version).

So, in summary:
- get_coverages() and get_normalized coverages are unaffected by controls
- Control are ignored if you choose (normalization=NULL) or (normalization="RPM")
- Controls are subtracted (normalization="NCIS") or log2-ratio'ed (normalization="log2_ratio") if you choose these normalization methods
- The control-corrected coverages are available as the return value of $group_coverages()

Hope this helps,
-Eric

**Comment by twoshoes** [comment]
Thank you, that's really helpful, and thanks for getting back to me so fast.
I was just using RPM normalisation in group_coverages() and should've looked at the other options properly!

**Comment by ericfournier2** [comment]
No problem!

I'll be edit this issue to make it about improving the role of control sample in the vignette and documentation.
